hamilton/plugins/sklearn_plot_extensions.py

```diff
@@ -0,0 +1,53 @@
+import dataclasses
+from os import PathLike
+from typing import Any, Collection, Dict, Tuple, Type, Union
+
+
+try:
+    import sklearn.metrics
+except ImportError:
+    raise NotImplementedError("scikit-learn is not installed.")
+
+try:
+    from sklearn.exceptions import NotFittedError
+except ImportError:
+    raise NotImplementedError("scikit-learn is not installed.")
+
+try:
+    from matplotlib import pyplot as plt
+except ImportError:
+    raise NotImplementedError("matplotlib is not installed")
+
+from hamilton import registry
+from hamilton.io import utils
+from hamilton.io.data_adapters import DataSaver
+
+
+SKLEARN_PLOT_TYPES = [sklearn.metrics.ConfusionMatrixDisplay, sklearn.metrics.DetCurveDisplay, sklearn.metrics.PrecisionRecallDisplay, sklearn.metrics.PredictionErrorDisplay, sklearn.metrics.RocCurveDisplay]
+SKLEARN_PLOT_TYPES_ANNOTATION = Union[sklearn.metrics.ConfusionMatrixDisplay, sklearn.metrics.DetCurveDisplay, sklearn.metrics.PrecisionRecallDisplay, sklearn.metrics.PredictionErrorDisplay, sklearn.metrics.RocCurveDisplay]
+
+@dataclasses.dataclass
+class ConfusionMatrixPlotSaver(DataSaver):
+
+    path: Union[str, PathLike]
+
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        return SKLEARN_PLOT_TYPES
+
+    def save_plot(self, data: SKLEARN_PLOT_TYPES_ANNOTATION):
+        try:
+            plt.savefig(self.path, dpi=200)

```

Comment:
so for any arguments/keyword arguments we'll want to expose options via the "constructor variables" for them. e.g. handle people being able to exercise the functionality you would normally be able to have access to - https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.savefig.html.

Reply:
we'll also want to pull the correct value from `data` to be plotted.

Reply:
> we'll also want to pull the correct value from `data` to be plotted.

changed to `data.figure_.savefig`

Reply:
> so for any arguments/keyword arguments we'll want to expose options via the "constructor variables" for them. e.g. handle people being able to exercise the functionality you would normally be able to have access to - https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.savefig.html.

I've added the method to handle keyword args, let me know if anymore changes are required